console/console-init/ui/src/modules/iot-device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx

```diff
@@ -94,10 +98,35 @@ const DeviceDetailHeader: React.FunctionComponent<IDeviceDetailHeaderProps> = ({
     });
   };
 
+  const onConfirmConnectionType = () => {
+    let actionType;
+    if (connectiontype === DeviceConnectionType.CONNECTED_DIRECTLY) {
+      actionType = DeviceActionType.CHANGE_CONNECTION_TYPE_CONNECTED_DIRECTLY;
+    } else if (connectiontype === DeviceConnectionType.VIA_GATEWAYS) {
+      actionType = DeviceActionType.CHANGE_CONNECTION_TYPE_VIA_GATEWAYS;
+    }
+
+    if (actionType) {
+      dispatch({
+        type: types.SET_DEVICE_ACTION_TYPE,
+        payload: { actionType }
+      });
+    }
+  };
+
   const onSelectChangeConnectionType = () => {
     dispatch({
-      type: types.SET_DEVICE_ACTION_TYPE,
-      payload: { actionType: DeviceActionType.CHANGE_CONNECTION_TYPE }
+      type: types.SHOW_MODAL,
+      modalType: MODAL_TYPES.CHANGE_CONNECTION_TYPE,
+      modalProps: {
+        iconType: "danger",
+        confirmButtonLabel: "Change",
+        onConfirm: onConfirmConnectionType,
+        option: "change-button",
+        detail:
+          "This action can not be undone. After you changed the setting, all data under this setting will also be deleted and is uncoverable.",
```

Comment:
Please replace 'changed' with 'change' and it will be unrecoverable instead of uncoverable.